### PR TITLE
Fix permissions when packaging artifacts

### LINKED ISSUE #2004: dotnet execution fails with "invalid .deps" file error due to permissions issue in the tarball
## Steps to reproduce

1. Download the latest RHEL CLI tarball from https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-rhel-x64.latest.tar.gz

2. Unpack the file into **/opt/dotnet ** with sudo privilege. 
3. Set below environment settings to execute .NET Core

     export DOTNET_HOME=/opt/dotnet
     PATH=$PATH:/opt/dotnet
4. execute below

     dotnet --version

## Expected  behavior
     it should display the existing dotnet CLI version 

## Actual behavior
     invalid .deps file error

Changing the permission set manually to files under /opt/dotnet resolves the issue. 

## Environment data
`dotnet --version` output: 

1.0.0.001946

### PR DESCRIPTION
**Pull request by ellismg**
The file permissions in our cross platform packages were all over the
place. Problems included:

- Executable text files
- Executable MSIL files
- Files not readable by world or group

In addition to just looking bad, it could cause problems in cases where
someone takes the tarballs and copies them to a global location (as
root) because now the deps file was not readable by non root users.

Add an additional step when setting everything up to put sensible
permissions on everything before building os specific packages and
tarballs.

Fixes #2004

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2015)

### CODE REVIEW

**Comment by ellismg** [comment]
@mellinoe Can you take a look as well?

File: scripts/dotnet-cli-build/PackageTargets.cs [hunk 1]
```diff
@@ -242,5 +247,11 @@ private static void CreateTarBallFromDirectory(string directory, string artifact
                 .Execute()
                 .EnsureSuccessful();
         }
+
+        private static void FixPermissions(string directory)
+        {
+            FS.ChmodAll(directory, "*", "644");
```
Comment (mellinoe) [inline scripts/dotnet-cli-build/PackageTargets.cs]: Can you put a comment on what "644" means

Comment (ellismg) [inline scripts/dotnet-cli-build/PackageTargets.cs]: Can do.

**Comment by mellinoe** [comment]
LGTM

File: scripts/dotnet-cli-build/PackageTargets.cs [hunk 2]
```diff
@@ -97,6 +100,8 @@ public static BuildTargetResult CopySharedFxLayout(BuildTargetContext c)
 
             Directory.CreateDirectory(sharedFxRoot);
             Utils.CopyDirectoryRecursively(Path.Combine(Dirs.Stage2, "shared"), sharedFxRoot, true);
+            FixPermissions(sharedFxRoot);
```
Comment (brthor) [inline scripts/dotnet-cli-build/PackageTargets.cs]: Should we fix these permissions before copying them to stage2? That way the stage2 directory has the correct perms too

**Comment by brthor** [comment]
Ignore last comment.

LGTM :shipit:

**Comment by ellismg** [comment]
@dotnet-bot test this please

**Comment by ellismg** [comment]
@dotnet-bot test Ubuntu x64 Debug Build please

**Pull request merged**